### PR TITLE
WIP: Do not merge, Dockerfile to create Jetson TX1 and TX2 compatible builds.

### PR DESCRIPTION
**Pull request by KellenSunderland**
## Description ##
Builds are ARM64 with CUDA 8 support.  Dependencies based on Jetpack 3.1.

### Essentials ###
- [x] Passed code style checking (`make lint`)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [x] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- [x] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### CODE REVIEW

**Comment by marcoabreu** [comment]
Are you sure it's cuda 9? The dockerfile states cuda 8 as base image

File: docker_multiarch/Dockerfile.build.jetson [hunk 1]
```diff
@@ -0,0 +1,74 @@
...
+
+# Create a binary wheel for easy installation.
+# When using tool.py output will be in the jetson folder.
+# Scp the .whl file to your target device, and install via
+# pip install
+WORKDIR /work/mxnet/python
+RUN python setup.py bdist_wheel
+
+# Copy build artifacts to output folder for tool.py script
+RUN mkdir -p /work/build & cp dist/mxnet-1.0.1-py2-none-any.whl /work/build && cp ../lib/* /work/build
```
Comment (marcoabreu) [inline docker_multiarch/Dockerfile.build.jetson]: Could the 1.0.1 in the filename be replaced with a dynamic variable in order to be compatible with version changes in the future?

File: docker_multiarch/arm.crosscompile.mk [hunk 2]
```diff
@@ -159,4 +159,4 @@ USE_CPP_PACKAGE = 0
 # whether to use sframe integration. This requires build sframe
 # git@github.com:dato-code/SFrame.git
 # SFRAME_PATH = $(HOME)/SFrame
-# MXNET_PLUGINS += plugin/sframe/plugin.mk
\ No newline at end of file
+# MXNET_PLUGINS += plugin/sframe/plugin.mk
```
Comment (marcoabreu) [inline docker_multiarch/arm.crosscompile.mk]: Whitespace change, maybe revert

**Comment by larroy** [comment]
Should we build with CMake?

File: docker_multiarch/arm.crosscompile.mk (hunk 2 above)
Comment (KellenSunderland) [reply docker_multiarch/arm.crosscompile.mk]: Should be a whitespace at the end of files, but I can put this in another commit.

File: docker_multiarch/Dockerfile.build.jetson (hunk 1 above)
Comment (KellenSunderland) [reply docker_multiarch/Dockerfile.build.jetson]: Good catch.  Should certainly be *.whl

**Comment by KellenSunderland** [comment]
@larroy I'd prefer to keep it with normal Makefiles for the time being as that's what the other Dockerfiles are using.  I had some issues building with CMake as well, for example cuda configuration right now is determined by the host GPU, which isn't exactly what you want when cross compiling.  We could get it to work with CMake, but since both build systems are valid I choose traditional make.

File: docker_multiarch/Dockerfile.build.jetson [hunk 3]
```diff
@@ -0,0 +1,74 @@
+# -*- mode: dockerfile -*-
+# dockerfile to build libmxnet.so for armv7
```
Comment (marcoabreu) [inline docker_multiarch/Dockerfile.build.jetson]: change to ARMv8?

File: docker_multiarch/Dockerfile.build.jetson [hunk 4]
```diff
@@ -0,0 +1,74 @@
...
+
+FROM nvidia/cuda:8.0-cudnn5-devel as cudabuilder
```
Comment (marcoabreu) [inline docker_multiarch/Dockerfile.build.jetson]: CUDA8 or 9?

Comment (KellenSunderland) [reply docker_multiarch/Dockerfile.build.jetson]: It should be 8.

File: docker_multiarch/Dockerfile.build.jetson (hunk 3 above)
Comment (KellenSunderland) [reply docker_multiarch/Dockerfile.build.jetson]: nice, updated.

**Comment by marcoabreu** [comment]
I wouldn't mind seeing this added to the Jenkinsfile during build-stage

File: Jenkinsfile [hunk 5]
```diff
@@ -281,6 +281,16 @@ try {
         }
       }
     },
+    'GPU: Jetson - ARM': {
```
Comment (marcoabreu) [inline Jenkinsfile]: Rename to armv8

File: Jenkinsfile [hunk 6]
```diff
@@ -281,6 +281,16 @@ try {
...
+      node('mxnetlinux-cpu') {
+        ws('workspace/jetson') {
+          init_git()
+          timeout(time: max_time, unit: 'MINUTES') {
+            sh "python docker_multiarch/tool.py -a jetson"
```
Comment (marcoabreu) [inline Jenkinsfile]: We should reconsider our directory structure. At the moment, all CI related dockerfiles are in the ci_build dir, but I also don't think that these dockerfiles should be exclusively for CI.

Comment (KellenSunderland) [reply Jenkinsfile]: Agree, we should merge these all in the future and get rid of ci_build.sh

File: Jenkinsfile (hunk 5 above)
Comment (KellenSunderland) [reply Jenkinsfile]: Please do not comment until WIP removed.

**Comment by KellenSunderland** [comment]
Iterating on this one in a private CI for a while.  Will reopen when stable.

**Pull request closed without merging**